lib/libutee/arch/arm/utee_mcount_a32.S

```diff
@@ -57,6 +57,16 @@ FUNC __ftrace_return, :
 	ldmia		sp!, {r0-r3}
 	bx		lr
 END_FUNC __ftrace_return
+
+FUNC read_cntpct, :
```

Comment:
This is arguably not the best place to have these functions, because they could be useful outside of ftrace. So they could have their own `CFG_` and be moved somewhere else.
Not a big deal however, could be done later.

Reply:
Agree.

Reply:
They are already generated and defined in `<out-dir>/core/include/generated/arm32_sysreg.h` and as assembly macros in `arm32_sysreg.S`

Reply:
@jenswi-linaro excellent :+1:

Reply:
@jenswi-linaro These seems to used for `core` build only and not exported for TA library builds. I get following error when I tried to use them:

```
  AS      out/arm/ta_arm32-lib/libutee/arch/arm/utee_mcount_a32.o
lib/libutee/arch/arm/utee_mcount_a32.S:7:10: fatal error: generated/arm32_sysreg.S: No such file or directory
 #include <generated/arm32_sysreg.S>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
mk/compile.mk:147: recipe for target 'out/arm/ta_arm32-lib/libutee/arch/arm/utee_mcount_a32.o' failed
make[1]: *** [out/arm/ta_arm32-lib/libutee/arch/arm/utee_mcount_a32.o] Error 1
make[1]: Leaving directory '/home/sumit/build/optee/optee_os'
common.mk:396: recipe for target 'optee-os-common' failed
make: *** [optee-os-common] Error 2
```

Reply:
Yes, perhaps we can generate a subset with the registers which can be accessed from EL0 into `<out>/include/generated/arm32_user_sysreg.h`. One way could be to copy the relevant lines from `core/arch/arm/kernel/arm32_sysreg.txt`.

For arm64 it's a bit easier due to the convenient assembly mnemonic, there we can just hand code all the EL0 accessible registers in `lib/libutee/include/arm64_user_sysreg.h`.

Reply:
Ok, let me try to create a `sysreg` subset for user access.